homework-submission/react-hyf/react-hyf-client/src/Components/Homework-w03/index.jsx

```diff
@@ -0,0 +1,61 @@
+import React, { useState } from 'react';
+import DataFetching from '../fetchdata.js';
+import ProfileCard from './Components/ProfileCard';
+import HyfButton from '../Homework-w02/components/HyfButton/button';
+
+function Week03() {
+  const defaultURL = 'https://uinames.com/api/?ext';
+  const maleURL = 'https://uinames.com/api/?ext&&gender=male';
+  const femaleURL = 'https://uinames.com/api/?ext&&gender=female';
+  const [URL, setURL] = useState(defaultURL);
+  const { loading, results, error, setRandomNumber } = DataFetching(URL);

```

Comment:
Because of your abstraction of the data fetching implementation these variables all sound a bit generic. The following would make more clear what's happening:

```js
const { loading: isLoadingUser, results: user, error, setRandomNumber } = DataFetching(URL);
```

Reply:
fixed in the homework for week 04 :)